### ISSUE TITLE
File already exists when picking a video from the image library in iOS

### ISSUE THREAD

**Issue by qimingweng** [marcshilling/react-native-image-picker#207]
![img_6853](https://cloud.githubusercontent.com/assets/8117770/15369672/b25e6268-1d01-11e6-83d5-a8ac8292a006.PNG)

I'm getting this error with this code

```js
ImagePickerManager.launchImageLibrary(
  {
    allowsEditing: false,
    noData: false,
    mediaType: 'video',
  },
  callback
);
```

**Comment by marcshilling** [comment]
What is `CreateNoteView.js`? Can I see line 111? It seems like you're throwing that `console.error`. `react-native-image-picker` does not throw any `console.error`s

**Comment by qimingweng** [comment]
@marcshilling Sorry, I'm just logging error returned in the response object if an error exists

Interestingly enough, selecting a video file from the simulator doesn't return a similar error

**Comment by qimingweng** [comment]
Basically, in the callback I do
```js
if (response.error) {
  console.error(response.error);
  return;
}
```

**Comment by marcshilling** [comment]
I see. Well, the response is not lying. You already have that file there. If you delete the app and reinstall, I'm guessing you can select the video successfully. Is that correct? However, I could change that code so that it overwrites existing files...

**Comment by marcshilling** [comment]
@qimingweng should be fixed in `v0.18.16`. Can you confirm?

**Comment by qimingweng** [comment]
@marcshilling It's a different error now "The file doesn't exist."

ps. you put my open source response times to shame

**Comment by qimingweng** [comment]
This, btw, is after deleting the app and reinstalling it too, in case it had anything to do with the temp documents or documents directory

**Comment by qimingweng** [comment]
I have a question, why do

```objc
NSURL *videoURL = info[UIImagePickerControllerMediaURL];
NSURL *videoDestinationURL = [NSURL fileURLWithPath:path];

NSFileManager *fileManager = [NSFileManager defaultManager];

// Delete file if it already exists
if ([fileManager fileExistsAtPath:videoDestinationURL.path]) {
    [fileManager removeItemAtURL:videoDestinationURL error:nil];
}

NSError *error = nil;
[fileManager moveItemAtURL:videoURL toURL:videoDestinationURL error:&error];
if (error) {
    self.callback(@[@{@"error": error.localizedFailureReason}]);
    return;
}

[response setObject:videoDestinationURL.absoluteString forKey:@"uri"];
```

instead of just


```objc
NSURL *videoURL = info[UIImagePickerControllerMediaURL];
[response setObject:videoURL.absoluteString forKey:@"uri"];
```

**Comment by marcshilling** [comment]
Wow, I'm just realizing this...it looks like iOS automatically puts a copy in the tmp directory for you...

Will have this fixed shortly

**Comment by qimingweng** [comment]
I've just vendored the code for now but will take a look at your fix when you have it ready, is it different from what I proposed?

**Comment by marcshilling** [comment]
In your case, no. But I need to account for if a user provides `storageOptions` to store the file in a different path. Almost done.

**Comment by marcshilling** [comment]
@qimingweng ok, definitely fixed in `v0.18.17`. Tested with and without `storageOptions`, from library and from camera. Thanks for helping me debug this.
